Sources/TuistGenerator/Generator/SchemesGenerator.swift

```diff
@@ -119,9 +119,24 @@ final class SchemesGenerator: SchemesGenerating {
         if fileHandler.exists(sharedPath) { try fileHandler.delete(sharedPath) }
     }
     
-    private func createDefaultScheme(target: Target, project: Project, buildConfiguration: String) -> Scheme {
+    func createDefaultScheme(target: Target, project: Project, buildConfiguration: String, graph: Graphing) -> Scheme {
         let targetReference = TargetReference.project(path: project.path, target: target.name)
-        let testTargets = target.product.testsBundle ? [TestableTarget(target: targetReference)] : []
+        
+        let testTargets: [TestableTarget]
+        
+        if target.product.testsBundle {
+            testTargets = [TestableTarget(target: targetReference)]
+        } else if let targetNode = graph.findTargetNode(path: project.path, name: target.name) {
+            // Find all test targets that depend on this target
+            testTargets = graph.targets(at: project.path)
```

Comment:
What about moving this logic to the `Graph` model? Something along the lines of:

```swift
func testableDependants(of: TargetNode) -> [TargetNode]
```

Reply:
@pepibumur updated!